### PR TITLE
Allow sharing questions from the example course

### PR DESCRIPTION
**Pull request by nwalters512**
This PR allows questions from the example course to be shared. We're going to use this to make "template" questions available for new users to try out immediately after signing up for a PrairieLearn account.

Crucially, we do _not_ allow template questions to be imported into courses via the sharing name + QID. We don't want to hold ourselves to the stability guarantees that would be required for shared questions. Currently, this is done via a hardcoded check for `pl_courses.example_course`, but in the future, we'll make this configurable on a per-question (and maybe per-course) basis.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ejs [hunk 1]
```diff
@@ -100,7 +100,7 @@
                       <span class="badge color-gray1"><%= sharing_set.name %></span>
                     <% }); %>
                   <% } %>
-                  <% if (authz_data.has_course_permission_own && !course.example_course) { %>
+                  <% if (authz_data.has_course_permission_own) { %>
```
Comment (mwest1066) [inline apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ejs:103]: Is this the key change in this file? To allow owners of the example course to turn on sharing?

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ejs (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ejs:103]: Correct; the other ones were just opportunistic formatting tweaks. Without this, there's no way for anyone to actually mark an example course question as shared.

Comment (mwest1066) [reply apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ejs:103]: Perfect, thanks!

Comment (SethPoulsen) [reply apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ejs:103]: I went to go check if we needed to make this same change on the backend in `instructorQuestionSettings.js`, but I guess we never had that check there (oops).

**Review by SethPoulsen — APPROVED** [review]
Looks good.

**Pull request merged**